src/main/java/com/alipay/oceanbase/hbase/OHTable.java

```diff
@@ -540,38 +543,175 @@ public boolean[] exists(List<Get> gets) throws IOException {
         return objectResults;
     }
 
+    private void compatOldServerBatch(final List<? extends Row> actions, final Object[] results, BatchError batchError)
+            throws Exception {
+        for (Row row : actions) {
+            if (!(row instanceof Put) && !(row instanceof Delete)) {
+                throw new FeatureNotSupportedException(
+                        "not supported other type in batch yet,only support put and delete");
+            }
+        }
+        List<Integer> resultMapSingleOp = new LinkedList<>();
+        List<Row> puts = new LinkedList<>();
+        for (int i = 0; i < actions.size(); i++) {
+            BatchOperationResult tmpResults;
+            BatchOperation batch = null;
+            if (actions.get(i) instanceof Put && ((Put)actions.get(i)).getFamilyCellMap().size() == 1) {
+                puts.clear();
+                Put put = (Put) actions.get(i++);
+                String family = Bytes.toString(put.getFamilyCellMap().firstKey());
+                // aggregate put
```

Comment:
this function is too long, can we wrapper this put aggregation process into a internal function

Reply:
fixed